File: src/my_cpp_pkg/src/add_two_ints_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using std::placeholders::_1;
// using std::placeholders::_2;

class AddTwoIntsServerNode : public rclcpp::Node
{
public:
   AddTwoIntsServerNode() : Node("add_two_ints_server")
   {
        server_ = this->create_service<example_interfaces::srv::AddTwoInts>(
            "add_two_ints",
            std::bind(&AddTwoIntsServerNode::callbackAddTwoInts, this, _1, std::placeholders::_2));
            RCLCPP_INFO(this->get_logger(), "AddTwoIntsServerNode initialized and server service started" );
   }
 
private:
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;

    void callbackAddTwoInts(const example_interfaces::srv::AddTwoInts_Request::SharedPtr request, 
                            const example_interfaces::srv::AddTwoInts_Response::SharedPtr response)
    {
        response->sum = request->a + request->b;
        RCLCPP_INFO(this->get_logger(), "%ld + %ld = %ld", request->a, request->b, response->sum );
    }
};
 
int main(int argc, char **argv)
{
   rclcpp::init(argc, argv);
   auto node = std::make_shared<AddTwoIntsServerNode>();
   rclcpp::spin(node);
   rclcpp::shutdown();
   return 0;
}

File: src/my_py_pkg/my_py_pkg/add_two_ints_server.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
 
 
class AddTwoIntsServerNode(Node):
   def __init__(self):
       super().__init__("add_two_ints_server")
       self.server_ = self.create_service(AddTwoInts, "add_two_ints", self.callback_add_two_ints)
       self.get_logger().info("AddTwoIntsServerNode initialized")
    
   def callback_add_two_ints(self, request, response):
       response.sum = request.a  + request.b
       self.get_logger().info("request.a " + str(request.a) + " + request.b " +str(request.b) + " = " + str(response.sum))
       return response
       
 
 
def main(args=None):
   rclpy.init(args=args)
   node = AddTwoIntsServerNode()
   rclpy.spin(node)
   rclpy.shutdown()
 
 
if __name__ == "__main__":
   main()